NO TICKET [update] add Get all Bulk Pricing Rules endpoint

bc-traciporter opened the pull request:
<!-- Ticket number or summary of work -->
# NO TICKET

## What changed?

Add get all bulk pricing rule endpoint
<!-- Provide a bulleted list in the present tense -->
* 

## Release notes draft

Bug Fix
Add missing endpoint
<!-- Provide an entry for the release notes using simple, conversational language. Don't be too technical. Explain how the change will benefit the merchant and link to the feature.

Examples:
* The newly-released [X feature] is now available to use. Now, you’ll be able to [perform Y action].
* We're happy to announce [X feature], which can help you  [perform Y action].
* [X feature] helps you to create [Y response] using the [Z query parameter]. Now, you can deliver [ex, localized shopping experiences for your customers].
* Fixed a bug in the [X endpoint]. Now the [Y field] will appear when you click [Z option]. -->
* 

## Anything else?
<!-- Add related PRs, salient notes, additional ticket numbers, etc. -->

ping {names}

bc-romanliukshyn commented on reference/catalog/products_catalog.v3.yml:
```diff
@@ -3424,6 +3424,74 @@ paths:
+                    total_pages: 1
+                    links:
+                      current: '?page=1&limit=50'
+         '404':
```
I'm not sure if we can have a 404 on the batch endpoint, but I would like to hear from the Pricing team.
cc @megdesko

megdesko replied:
We typically don't return 404, we return 200 OK with an empty array in `data` on batch endpoints.

bc-romanliukshyn commented on reference/catalog/products_catalog.v3.yml:
```diff
@@ -3424,6 +3424,74 @@ paths:
+      tags:
+        - Bulk pricing rules
+      summary: Get all Bulk Pricing Rules
+      description: Returns all *Bulk Pricing Rule*. Optional parameters can be passed in.
```
```suggestion
      description: Returns all *Bulk Pricing Rules*. Optional parameters can be passed in.
```

bc-romanliukshyn commented on reference/catalog/products_catalog.v3.yml:
```diff
@@ -3424,6 +3424,74 @@ paths:
+      description: Returns all *Bulk Pricing Rules*. Optional parameters can be passed in.
+      operationId: getAllBulkPricingRules
+      parameters:
+        - $ref: '#/components/parameters/IncludeFieldsParam'
```
Hi @bc-traciporter , this leads to show wrong fields for this endpoint:
![image](https://github.com/user-attachments/assets/b2a83436-23d4-455f-98b3-585f71516d65)
These are for product, not for pricing rule.

bc-traciporter replied:
@bc-romanliukshyn should I also remove 
- $ref: '#/components/parameters/ExcludeFieldsParam'
![Screenshot 2025-02-06 at 9 30 30 AM](https://github.com/user-attachments/assets/d50e677c-79f8-4012-aeb4-7834eef25640)

bc-romanliukshyn replied:
exclude_fields should be available:
![image](https://github.com/user-attachments/assets/949c8584-146a-4a13-8b38-1df976d6724b)

bc-romanliukshyn commented on reference/catalog/products_catalog.v3.yml:
```diff
@@ -3424,6 +3424,74 @@ paths:
+                            readOnly: true
+                        required:
+                          - id
+                      - $ref: '#/components/schemas/bulkPricingRule_Full'
```
![image](https://github.com/user-attachments/assets/4a33aa29-cbbb-4f11-9916-dda27d2327f1)
Fields should not be marked as required in the response.

bc-romanliukshyn commented on reference/catalog/products_catalog.v3.yml:
```diff
@@ -3424,6 +3424,74 @@ paths:
+          content:
+            application/json:
+              schema:
+                title: 'Get All Bulk Pricing Rule Response'
```
```suggestion
                title: 'Get All Bulk Pricing Rules Response'
```

bc-traciporter commented:
Hi @bc-romanliukshyn I removed the required fields from the response bodies. @megdesko said to remove the 404 error response for the bulk endpoint. I did that. Also I removed the `include_fields` query parameter. I am just wondering if I need to also remove the `exclude_fields` parameter.

bc-traciporter commented:
@bc-romanliukshyn I added the include_fields query params

bc-romanliukshyn commented:
> @bc-romanliukshyn I added the include_fields query params

thanks @bc-traciporter ! Not sure if you want to do this in scope of this PR, but there is the same issue with the `include_fields` param in the Get a Bulk Pricing Rule endpoint. Everything else looking good to me.

bc-traciporter commented:
> > @bc-romanliukshyn I added the include_fields query params
> 
> thanks @bc-traciporter ! Not sure if you want to do this in scope of this PR, but there is the same issue with the `include_fields` param in the Get a Bulk Pricing Rule endpoint. Everything else looking good to me.

@bc-romanliukshyn This is a quick change. So the include fields should be the same as the Get all bulk pricing rule?

bc-romanliukshyn commented:
> > > @bc-romanliukshyn I added the include_fields query params
> > 
> > 
> > thanks @bc-traciporter ! Not sure if you want to do this in scope of this PR, but there is the same issue with the `include_fields` param in the Get a Bulk Pricing Rule endpoint. Everything else looking good to me.
> 
> @bc-romanliukshyn This is a quick change. So the include fields should be the same as the Get all bulk pricing rule?

Yep, thank you. Approved
